### DISCUSSION TITLE
Is there a way to check permissions for the current credentials?

### DISCUSSION THREAD

**Discussion by tomkerkhove** [kubernetes-client/csharp#1058] [Q&A] [answered]
Is there a way to check permissions for the current credentials?

**Comment by tg123** [comment, marked as answer]
client.AuthorizationV1.CreateSelfSubjectAccessReview

https://kubernetes.io/docs/reference/kubernetes-api/authorization-resources/self-subject-access-review-v1/

**Reply by tomkerkhove** [reply]
That requires you to go through all the resources to check the permission. I presume there is no other way then?

**Reply by tg123** [reply]
what kind of permission do you want to check

**Reply by tomkerkhove** [reply]
I want to determine what the permissions are that I currently have to assess if the workload doesn't have too much permissions (without wanting to loop things)

**Reply by tomkerkhove** [reply]
Or if I want to loop things, I'd like to assess all the things easily to build a report

**Reply by tg123** [reply]
what if you have a crd?

**Reply by tomkerkhove** [reply]
Are you asking if I want to know about permissions for using a CRD or if I can use a CRD to get the info?

If the former, no need to check permissions for CRDs.

**Reply by tg123** [reply]
hmmm seems you have to check every single resource you wish to generate the full report

**Reply by tomkerkhove** [reply]
No problem, thanks for the help!
